File: api-artist-workspace/src/main/java/org/example/workspace/repository/UserRepository.java
package org.example.workspace.repository;

import org.example.workspace.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByLoginIdAndIsDeletedFalse(String username);

    Optional<User> findByEmailAndIsDeletedFalse(String username);

    Optional<User> findByIdAndIsDeletedFalse(Long id);


    Optional<User> findByIdAndEmailAndIsDeletedFalse(Long userId, String userEmail);
}

File: level-up-backend-gpt/level2-book-management-system/src/main/kotlin/org/example/level2bookmanagementsystem/base/repository/UserRepository.kt
package org.example.level2bookmanagementsystem.base.repository

import org.example.level2bookmanagementsystem.security.entity.TbUser
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<TbUser, Long> {
    fun findByUserLoginId(userId: String): TbUser?
}